app/models/course/assessment/answer/text_response.rb

```diff
@@ -49,6 +49,16 @@ def assign_params(params)
     self.files = params[:files] if params[:files]
   end
 
+  def compare_answer(other_answer)
+    same_text = answer_text == other_answer.answer_text
+    same_attachment_length = attachments.length == other_answer.attachments.length
+    answer_filename_attachment = attachments.pluck(:name, :attachment_id).map { |elem| elem.join('#') }
```

Comment:
Can I check:

- I think attachment id definitely works, but I'm thinking about the case where I upload one file, delete it, then reupload it. Should we bother hashing the file before comparing the hashes?
- Should we sort the ids before joining? In case two answers have the same ids but in different orders.

Reply:
- When a file is uploaded, it is hashed and kept in the attachment table. If it is deleted, the file is actually not removed and so the next time the same file is uploaded, it will be rehashed and since the hash is the same, it will not create a new record in the attachment table. Instead, it will use the old one. As such, the attachment id is the same.
- The order does not matter here since we are comparing the set below (see line 58 - https://github.com/Coursemology/coursemology2/pull/4190/files#diff-c2181849ffadc500da87a8855cdced098b6139af18d72d9f53c4926584a46b2eR58)

Reply:
Ah I see, ok. Thanks for clarifying, misunderstood the purpose of the join.